runtime/cudaq/platform/default/rest_server/helpers/JsonConvert.h

```diff
@@ -150,10 +150,18 @@ class RestRequest {
   // This needs to be bumped whenever a breaking change is introduced.
   // e.g., adding/removing non-optional fields, changing field names, etc.
   static constexpr std::size_t SCHEMA_VERSION = 1;
+  // Version string identifying the client version.
+  static inline const std::string CUDA_QUANTUM_VERSION = []() {
+    std::stringstream ss;
+    ss << "CUDA Quantum Version " << cudaq::getVersion() << " ("
+       << cudaq::getFullRepositoryVersion() << ")";
+    return ss.str();
+  }();
 
 public:
   RestRequest(ExecutionContext &context)
-      : executionContext(context), version(SCHEMA_VERSION) {}
+      : executionContext(context), version(SCHEMA_VERSION),
+        client_version(CUDA_QUANTUM_VERSION) {}
```

Comment:
Sorry if I missed it, but is this full string being checked someplace here?

If so, it doesn't seem like a good idea necessarily. Clients and servers may get slightly out of synch with one another, and it will frustrate people to mandate they be in lockstep.

Reply:
> Sorry if I missed it, but is this full string being checked someplace here?

I believe it's just being logged (for future debugging considerations).

Reply:
Yes, this is only for logging purposes at this stage so that we know exactly the version of the frontend client.